Firmware/Marlin_main.cpp

```diff
@@ -6628,8 +6624,13 @@ if((eSoundMode==e_SOUND_MODE_LOUD)||(eSoundMode==e_SOUND_MODE_ONCE))
         else
         {
           #ifdef FILAMENTCHANGE_ZADD
-            z_shift= FILAMENTCHANGE_ZADD ;
-            if(current_position[Z_AXIS] < 25) z_shift+= 25 ;
+			static_assert(Z_MAX_POS < (255 - FILAMENTCHANGE_ZADD), "Z-range too high, change the following code from uint8_t to uint16_t");
+			// avoid floating point arithmetics when not necessary - results in shorter code
+			uint8_t ztmp = uint8_t( current_position[Z_AXIS] );
+			if(ztmp < uint8_t(25)){
+				z_shift = uint8_t(25) - ztmp; // make sure to be at least 25mm above the heat bed
+			}
+			z_shift += FILAMENTCHANGE_ZADD ; // always move above printout
```

Comment:
It would be nice to extract this snipped into function, so it can be checked, if it does what is intended.